docs/source/topics/data_types.rst

```diff
@@ -0,0 +1,508 @@
+.. _topics:data_types:
+
+**********
+Data types
+**********
+
+AiiDA already ships with a number of useful data types.
+This section details the most common, and some handy features/functionalities to work with them.
+
+.. _topics:data_types:core:
+
+Core data types
+===============
+
+Here follows a short summary of common data types already provided with AiiDA.
+This list is not complete, see also inside :py:mod:`aiida.orm.nodes.data` for the list of all available plugins.
+
+We also mention, when relevant, what is stored in the database (as attributes, so that it can be easily queried e.g. with the :ref:`QueryBuilder <how-to:data:find>`) and what is stored in the file repository (providing access to the file contents, but not efficiently queryable: this is useful for e.g. big data files that don't need to be queried for).
+
+For all data types, you can follow the link to the data class to read more about the methods provided, how to access them, and so on.
+
+If you need to work with some specific type of data, first check the list of data types/plugins below, and if you don't find what you need, give a look to :ref:`how to write a new data plugin <how-to:data:plugin:create>`.
+
+.. _topics:data_types:core:base:
+
+Base types
+----------
+
+In the :py:mod:`aiida.orm.nodes.data.base` module there are a number of useful classes that wrap base python datatypes (like :py:class:`~aiida.orm.nodes.data.int.Int`, :py:class:`~aiida.orm.nodes.data.float.Float`, :py:class:`~aiida.orm.nodes.data.str.Str`, ...).
+These are automatically loaded with the verdi shell, and also directly exposed from :py:mod:`aiida.orm`.
+These classes are particularly useful when you need to provide a single parameter to e.g. a :py:class:`~aiida.engine.processes.functions.workfunction`.
+
+Each of these classes can most often be used transparently (e.g. you can sum two :py:class:`~aiida.orm.nodes.data.int.Int` objects, etc.). If you need to access the bare value and not the whole AiiDA class, use the ``.value`` property.
+
+In the same module, there is also a :py:class:`~aiida.orm.nodes.data.list.List` class to store a list of base data types.
```

Comment:
Since for most classes you can retrieve the underlying value with `.value`, but for lists and dicts it's different, it would be good to add a small paragraph on this

Reply:
I agree, but just out of curiosity: why is this different? Also, should we perhaps consider changing the `__str__` method of e.g. `List` in this case, as now this can be somewhat confusing:

```Python
In [1]: l = List(list=[1, 2])

In [2]: l
Out[2]: <List: uuid: fc6335b7-69d9-4d65-bc1a-9adb5242a69f (unstored) value: [1, 2]>

In [3]: l.value
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-aa957b8aeff4> in <module>
----> 1 l.value

AttributeError: 'List' object has no attribute 'value'
```

Reply:
Another question I had regarding the `List` and `Dict` classes is the fact that the user cannot simply initialize e.g. a `List` using:

```Python
l = List([1, 2, 4])
```

Instead, he or she has to specify the `list` argument with its key:

```Python
l = List(list=[1, 2, 4])
```

It's also not immediately clear from the constructor signature or docstring that you have to initialize the `List` this way:

```Python
In [1]: List?
Init signature: List(**kwargs)
Docstring:      `Data` sub class to represent a list.
Init docstring:
:param backend_entity: the backend model supporting this entity
:type backend_entity: :class:`aiida.orm.implementation.entities.BackendEntity`
File:           ~/envs/aiida-docs/code/aiida-core/aiida/orm/nodes/data/list.py
Type:           AbstractNodeMeta
Subclasses:    
```

Is there a good reason for this? I.e. why don't we change the current constructor:

```Python
def __init__(self, **kwargs):
    data = kwargs.pop('list', list())
    super().__init__(**kwargs)
    self.set_list(data)
```

to something like:


```Python
def __init__(self, input_list=None, **kwargs):
    input_list = input_list or list()
    super().__init__(**kwargs)
    self.set_list(input_list)
```

If so, we should at least add a docstring to the constructor. 😅

Reply:
On the first point: indeed, adding a .value to both List and Dict would be OK I think - do you want to open an issue, and/or ask this later at the meeting? I don't know if we also need to remove get_list(), get_dict(), maybe it's OK to keep both?

On the second point: 
- we cannot call it 'input_list' as this is backward-incompatible, we need to call it list (and of course this shadows python's list - but then one can write something like
  ```python
  import builtins
  list = list or builtins.list()
  ```
  (with a comment explaining what's going on ;-) )
- I think there was some historical reasons because, for *any* node, if you passed a kwarg `aaa`, it would have been passed to a method `set_aaa`. Probably this does not exist anymore? (pinging @sphuber) Anyway, maybe this is OK and what you suggest is OK, apart from the parameter name?

Reply:
Thanks @giovannipizzi! I opened issue #4495 to discuss these suggestions further.